test/constraints-validation.test.ts

```diff
@@ -0,0 +1,73 @@
+import 'source-map-support/register';
+import * as cdk from 'aws-cdk-lib';
+import * as blueprints from '../lib';
+import { BlueprintBuilder, ControlPlaneLogTypes } from '../lib';
+import { z, ZodError } from 'zod';
+
+const longName = 'eks-blueprint-with-an-extra-kind-of-very-long-name-that-is-definitely-not-going-to-work';
+
+const app = new cdk.App();
+
+const addOns: Array<blueprints.ClusterAddOn> = [
+  new blueprints.addons.MetricsServerAddOn(),
+  new blueprints.addons.ClusterAutoScalerAddOn(),
+  new blueprints.addons.VpcCniAddOn(),
+];
+
+function getConstraintsDataSet(): [BlueprintBuilder, ZodError][] {
+  let result: [BlueprintBuilder, ZodError][] = [];
+  const blueprint1 = blueprints.EksBlueprint.builder().addOns(...addOns).id(longName);
+  
+  const blueprint1ZodError = new z.ZodError([{
+    code: "too_big",
+    maximum: 63,
+    type: "string",
+    inclusive: true,
+    path: [],
+    message: "sdfasdf"
+  }]);
+
+  const blueprint2 = blueprints.EksBlueprint.builder().addOns(...addOns).enableControlPlaneLogTypes(ControlPlaneLogTypes.api).name(longName);
+  const blueprint2ZodError = new z.ZodError([{
+    //code: "too_small",
+    code: "too_big",
+    maximum: 63,
+    //minimum: 1,
+    type: "string",
+    inclusive: true,
+    path: [],
+    message: "sdfasdf"
+  }]);
+
+  result.push([blueprint1, blueprint1ZodError]);
+  result.push([blueprint2, blueprint2ZodError]);
+
+  return result;
+}
+
+test("For Each loop test.", () => {
+  getConstraintsDataSet().forEach((ex) => {
+    try{
+     ex[0].build(app, ex[0].props.id!);//be sure all tests built have a id defined before this!
+     expect(true).toBe(false);
+     
+    } catch(e) {
+      if(e instanceof z.ZodError){ //we are testing for code, maximum/minimum, and type
+        const eMax = (e.issues[0] as any).maximum;
```

Comment:
1. There is no need to have expression like `e.issues[0] as any` all over the code. Just extract a variable once. 
2. You can use a generic object comparison function to compare that all fields in object ex[1].issues[0] are equals to e.issues[0]. See more here: https://dmitripavlutin.com/how-to-compare-objects-in-javascript/#3-shallow-equality

Reply:
I figured there must be a better way to do that after I did it, just got it functional first, now for some fine tuning. I'll be sure to look into those things and make it better.